### DISCUSSION TITLE
Performance: Flows-Action Map() operation needed

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#14184] [Feature Requests]
### Describe the Bug

In order to read data and perform the same action based on the results we should have a map function. This would allow us to  create a new array of object-ids that can be used for the next action.

I know that we already can pass an array to a sub-flow and it will run it for each entry. But in this case the sub-flow needs to run once per object instead of updating all elements at once.

### To Reproduce

**_Based on the directus.cloud example project:_**

* Create a flow to read all data with the status "archived"
    ```
    {
      "filter": {
         "_and": [
              {
                   "status": {
          	       "_eq": "archived"
          	     }
              }
          ]
      }
    }
    ```
* Try to update the status all results

--> we can update the results one by one by triggering a sub-flow, but can't map all `response.id` to a new array and pass it to the update's ID field

### What version of Directus are you using?

9.12.2

### What version of Node.js are you using?

unknown (Directus cloud)

### What database are you using?

SQLite (Directus cloud)

### What browser are you using?

Chrome

### How are you deploying Directus?

Directus cloud

**Comment by rijkvanzanten** [comment]
I'm wondering how this ties into https://github.com/directus/directus/discussions/13944 as that might also allow for this 🙂

**Reply by Dominic-Marcelino** [reply]
This would definitely solve the problem!

As mentioned in the discussion, a custom script brings a whole new complexity with it, as well as the risk of easily break things unintentionally.

 Based on this, I see this it as two different things that could exist next to each other.

An action to create a new array out of an objects field is also more way more Basic and specific. This would e.g also allow to output / mail a list of titles, etc.

**Reply by rijkvanzanten** [reply]
Absolutely!

**Comment by Dominic-Marcelino** [comment]
For anyone that finds this by search: there's a new 'script' operator
